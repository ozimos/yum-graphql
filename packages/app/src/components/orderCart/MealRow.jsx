import React from 'react';
import PropTypes from 'prop-types';

const MealRow = ({
  id,
  title,
  price,
  quantity,
  setQuantity,
  removeFromCart
}) => {
  const subTotal = price * (quantity || 1);

  return (
    <React.Fragment>
      <tr className="slim">
        <td>{title}</td>
        <td>{price}</td>

        <td className="quantity-cell">

          <input
            className="meal-quantity"
            type="number"
            placeholder="1"
            onChange={event => setQuantity(event, id)}
            value={quantity}
            maxLength="4"
            max="99"
            min="1"
            step="1"
          />

        </td>

        <td>{subTotal}</td>
        <td>

          <button className="btn btn-cart" onClick={() => removeFromCart(id)}>
                &#10006;
          </button>

        </td>
      </tr>
    </React.Fragment>
  );
};

MealRow.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  quantity: PropTypes
    .oneOfType([PropTypes.string, PropTypes.number])
    .isRequired,
  removeFromCart: PropTypes.func.isRequired,
  setQuantity: PropTypes.func.isRequired,
};
export default MealRow;
