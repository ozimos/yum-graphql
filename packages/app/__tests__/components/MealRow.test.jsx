/* global React:false, shallow:false toJson:false */

import MealRow from '../../src/components/orderCart/MealRow';

describe('MealRow Component', () => {

  it('renders correctly', () => {
    const wrapperWithAllProps = shallow(<MealRow
      id="abc"
      title="abc"
      price={4}
      quantity={4}
      removeFromCart={jest.fn()}
      setQuantity={jest.fn()}
    />);
    expect(toJson(wrapperWithAllProps)).toMatchSnapshot();

    const wrapper = shallow(<MealRow
      id="abc"
      title="abc"
      price={4}
      removeFromCart={jest.fn()}
      setQuantity={jest.fn()}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper.find('.btn.btn-cart').simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper.find('input').simulate('change');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});

