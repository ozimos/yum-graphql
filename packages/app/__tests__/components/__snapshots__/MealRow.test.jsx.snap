// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealRow Component renders correctly 1`] = `
<React.Fragment>
  <tr
    className="slim"
  >
    <td>
      abc
    </td>
    <td>
      4
    </td>
    <td
      className="quantity-cell"
    >
      <input
        className="meal-quantity"
        max="99"
        maxLength="4"
        min="1"
        onChange={[Function]}
        placeholder="1"
        step="1"
        type="number"
        value={4}
      />
    </td>
    <td>
      16
    </td>
    <td>
      <button
        className="btn btn-cart"
        onClick={[Function]}
      >
        ✖
      </button>
    </td>
  </tr>
</React.Fragment>
`;

exports[`MealRow Component renders correctly 2`] = `
<React.Fragment>
  <tr
    className="slim"
  >
    <td>
      abc
    </td>
    <td>
      4
    </td>
    <td
      className="quantity-cell"
    >
      <input
        className="meal-quantity"
        max="99"
        maxLength="4"
        min="1"
        onChange={[Function]}
        placeholder="1"
        step="1"
        type="number"
      />
    </td>
    <td>
      4
    </td>
    <td>
      <button
        className="btn btn-cart"
        onClick={[Function]}
      >
        ✖
      </button>
    </td>
  </tr>
</React.Fragment>
`;

exports[`MealRow Component renders correctly 3`] = `
<React.Fragment>
  <tr
    className="slim"
  >
    <td>
      abc
    </td>
    <td>
      4
    </td>
    <td
      className="quantity-cell"
    >
      <input
        className="meal-quantity"
        max="99"
        maxLength="4"
        min="1"
        onChange={[Function]}
        placeholder="1"
        step="1"
        type="number"
      />
    </td>
    <td>
      4
    </td>
    <td>
      <button
        className="btn btn-cart"
        onClick={[Function]}
      >
        ✖
      </button>
    </td>
  </tr>
</React.Fragment>
`;

exports[`MealRow Component renders correctly 4`] = `
<React.Fragment>
  <tr
    className="slim"
  >
    <td>
      abc
    </td>
    <td>
      4
    </td>
    <td
      className="quantity-cell"
    >
      <input
        className="meal-quantity"
        max="99"
        maxLength="4"
        min="1"
        onChange={[Function]}
        placeholder="1"
        step="1"
        type="number"
      />
    </td>
    <td>
      4
    </td>
    <td>
      <button
        className="btn btn-cart"
        onClick={[Function]}
      >
        ✖
      </button>
    </td>
  </tr>
</React.Fragment>
`;
